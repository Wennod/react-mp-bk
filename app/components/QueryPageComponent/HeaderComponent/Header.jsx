import React from 'react';

import SearchComponent from './SearchComponent/SearchContainer';
import Logo from './LogoComponent/Logo';

class Header extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="header">
                <Logo className="header__logo" />
                <SearchComponent setQueryParam={this.props.setQueryParam} setQueryValue={this.props.setQueryValue} getMovies={this.props.getMovies} className="header__search-component" />
            </div>
        );
    }

}

export default Header;