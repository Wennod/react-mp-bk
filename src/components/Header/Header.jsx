import React from 'react';

import SearchComponent from './SearchComponent/SearchContainer';

import Style from './Header.style.scss';

class Header extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="header">
                <div className="header__logo-wraper">
                    <img className="header__logo" src="http://via.placeholder.com/100x25" alt="logo"/>        
                </div>
                <div className="header__search-wraper">
                    <SearchComponent 
                        setQueryParam={this.props.setQueryParam} 
                        setQueryValue={this.props.setQueryValue} 
                        getMovies={this.props.getMovies} 
                        className="header__search-component" 
                    />
                </div>
            </div>
        );
    }

}

export default Header;