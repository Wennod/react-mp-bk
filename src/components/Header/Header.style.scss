.header {
    display: flex;
    flex-flow: column;
    padding: 0 100px;
    flex-shrink: 0;

    /* temporal */
        background-color: ghostwhite;
    /* temporal */

    &__logo-wraper {
        margin: 20px 0;
    }
}