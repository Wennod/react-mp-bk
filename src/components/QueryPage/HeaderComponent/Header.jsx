import React from 'react';

import SearchComponent from './SearchComponent/SearchContainer';
import Logo from '../LogoComponent/Logo';

import Style from './Header.style.scss';

class Header extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="header">
                <div className="header__logo-wraper">
                    <Logo className="header__logo" />
                </div>
                <div className="header__search-wraper">
                    <SearchComponent setQueryParam={this.props.setQueryParam} setQueryValue={this.props.setQueryValue} getMovies={this.props.getMovies} className="header__search-component" />
                </div>
            </div>
        );
    }

}

export default Header;