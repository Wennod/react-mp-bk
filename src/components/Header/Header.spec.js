import React from 'react';
import Header from './Header';

describe('Header snapshot', () => {
    it('should match snapshot', () => {
        const tree = mount(<Header />);
        expect(tree).toMatchSnapshot();
    });
});