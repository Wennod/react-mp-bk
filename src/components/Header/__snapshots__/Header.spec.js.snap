// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot should match snapshot 1`] = `
<Header>
  <div
    className="header"
  >
    <div
      className="header__logo-wraper"
    >
      <img
        alt="logo"
        className="header__logo"
        src="http://via.placeholder.com/100x25"
      />
    </div>
    <div
      className="header__search-wraper"
    >
      <SearchComponentContainer
        className="header__search-component"
      >
        <SearchForm>
          <form
            className="search-form"
          >
            <h2>
              Find your movies
            </h2>
            <input
              className="search-form__input"
              id="fieldId"
              name="search"
              onChange={[Function]}
              type="text"
            />
            <div
              className="search-form__group"
            >
              <div
                className="search-form__filter"
              >
                <p
                  className="search-form__filter-title"
                >
                  Search By
                </p>
                <input
                  className="search-form__filter-toggler search-form__filter-toggler--t"
                  defaultChecked={true}
                  id="title"
                  name="searchCriteria"
                  onClick={[Function]}
                  type="radio"
                  value="title"
                />
                <input
                  className="search-form__filter-toggler search-form__filter-toggler-g"
                  id="genre"
                  name="searchCriteria"
                  onClick={[Function]}
                  type="radio"
                  value="genres"
                />
              </div>
              <input
                className="search-button"
                type="submit"
                value="SEARCH"
              />
            </div>
          </form>
        </SearchForm>
      </SearchComponentContainer>
    </div>
  </div>
</Header>
`;
